model/task_lifecycle.go

```diff
@@ -1127,6 +1127,36 @@ func updateBuildGithubStatus(b *build.Build, buildTasks []task.Task) error {
 	return b.UpdateGithubCheckStatus(buildStatus.status)
 }
 
+// checkUpdateBuildPRStatus checks if the build is coming from a PR, and if so
+// sends its updated status to GitHub to reflect the status of the build.
+func checkUpdateBuildPRStatus(b *build.Build) error {
+	if !evergreen.IsGitHubPatchRequester(b.Requester) {
+		return nil
+	}
```

Comment:
Is it worth having a check to verify that `b.Status` is an unfinished status, or is checking the patch status sufficient?

Reply:
Good catch. This is a symptom of moving around the logic `UpdatePatchStatus`, which had this issue in the first place. We want to be updating build statuses independently of the patch status changes. Changed the check at L1143 to `p.IsGithubPRPatch() && !evergreen.IsFinishedBuildStatus(b.Status)`